src/conversion/modelconversiondispatcher.js

```diff
@@ -252,6 +271,50 @@ export default class ModelConversionDispatcher {
 	}
 
 	/**
+	 * Fires events needed to handle rename conversion.
+	 *
+	 * Renaming is handled by:
+	 * * firing insertion conversion - for a new element (renamed clone of old element),
+	 * * firing move conversion - moving children of old element to new element,
+	 * * firing remove conversion - removing old element.
+	 *
+	 * This way two goals are achieved:
+	 * * no need to provide additional converters for renaming,
+	 * * contents of renamed element do not have to be re-rendered.
+	 *
+	 * @fires engine.conversion.ModelConversionDispatcher#event:rename
+	 * @fires engine.conversion.ModelConversionDispatcher#event:insert
+	 * @fires engine.conversion.ModelConversionDispatcher#event:move
+	 * @fires engine.conversion.ModelConversionDispatcher#event:remove
+	 * @param {engine.view.Element} element Renamed element.
+	 * @param {String} oldName Name of the renamed element before it was renamed.
+	 */
+	convertRename( element, oldName ) {
+		const consumable = new Consumable();
+		consumable.add( element, 'rename' );
+
+		// Callbacks added to `rename` event should either provide totally custom handling of rename conversion
+		// or create "fake element" in model that is needed to fire default rename conversion.
+		const data = { element, oldName };
+		this.fire( 'rename:' + element.name + ':' + oldName, data, consumable, this.conversionApi );
+
+		// Default callback should provide `data.fakeElement` model element that is needed for firing conversion.
+		if ( data.fakeElement ) {
```

Comment:
This will be simplified and will look like the other methods in `ModelConversionDispatcher`.